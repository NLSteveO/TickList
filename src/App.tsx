import { useState } from 'react';
import { AddRoute, Home } from './pages';

const PAGE_INDEX = 4;

const pageComponents: Record<string, React.ComponentType> = {
  addRoute: AddRoute,
  home: Home
};

const renderPage = (page: string) => {
  const PageComponent = pageComponents[page] || Home;
  return <PageComponent />;
};

const resetURL = () => {
  window.history.pushState({}, '', '/TickList');
  const event = new Event('popstate');
  window.dispatchEvent(event);
}

function App() {
  const splitURL = window.location.href.split('/');
  if (splitURL[PAGE_INDEX]) resetURL();

  const [page] = useState(splitURL[PAGE_INDEX]);

  return renderPage(page);
}

export default App;
