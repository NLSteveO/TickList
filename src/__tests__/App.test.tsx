import { describe, expect, it } from "vitest";
import App from '../App';

describe("App", () => {
  it("should render the page correctly", async () => {
    const result = App();
    expect(result).toMatchSnapshot();
  });
});
