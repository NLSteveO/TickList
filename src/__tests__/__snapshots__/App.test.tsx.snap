// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render the page correctly 1`] = `<Home />`;
